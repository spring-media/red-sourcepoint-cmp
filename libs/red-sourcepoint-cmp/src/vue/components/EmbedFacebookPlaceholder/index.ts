export { default as EmbedFacebookPlaceholder } from './EmbedFacebookPlaceholder.vue';
