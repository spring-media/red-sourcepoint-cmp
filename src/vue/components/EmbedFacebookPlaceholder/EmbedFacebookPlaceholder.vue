<template>
  <embed-social-networks-placeholder :privacy-manager-id="privacyManagerId" :vendor-id="vendorId">
    <template #header>
      <svg width="80" height="80" viewBox="0 0 80 80" fill="none" xmlns="http://www.w3.org/2000/svg">
        <path
          d="M73.3334 40.2035C73.3334 21.6815 58.4095 6.6665 40 6.6665C21.5905 6.6665 6.66669 21.6815 6.66669 40.2035C6.66669 56.9426 18.8561 70.8172 34.7917 73.3332V49.8979H26.3281V40.2035H34.7917V32.8149C34.7917 24.4097 39.7683 19.7669 47.3822 19.7669C51.0293 19.7669 54.8438 20.4219 54.8438 20.4219V28.6752H50.6406C46.5 28.6752 45.2084 31.2606 45.2084 33.9127V40.2035H54.4531L52.9753 49.8979H45.2084V73.3332C61.1439 70.8172 73.3334 56.9426 73.3334 40.2035Z"
          fill="#CED4DA"
        />
      </svg>
    </template>
    <template #headline> An dieser Stelle findest du Inhalte aus Facebook </template>
    <template #description>
      Um mit Inhalten aus Facebook und anderen sozialen Netzwerken zu interagieren oder diese darzustellen, brauchen wir
      deine Zustimmung.
    </template>
  </embed-social-networks-placeholder>
</template>

<script lang="ts">
import Vue from 'vue';
import { VENDOR_ID_FACEBOOK } from '../../../vendor-mapping';
import { EmbedSocialNetworksPlaceholder } from '../EmbedSocialNetworksPlaceholder';

type Props = {
  privacyManagerId: number;
  vendorId: string;
};

type NonNullish = Record<string, unknown>;

export default Vue.extend<NonNullish, NonNullish, NonNullish, Props>({
  name: 'EmbedFacebookPlaceholder',
  components: { EmbedSocialNetworksPlaceholder },
  props: {
    privacyManagerId: {
      type: Number,
      required: true,
    },
    vendorId: {
      type: String,
      default: VENDOR_ID_FACEBOOK,
    },
  },
});
</script>
