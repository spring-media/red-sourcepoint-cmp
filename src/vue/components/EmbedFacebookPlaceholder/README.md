# Placeholder component for Facebook embeds

<p>
  <img src="../../../../docs/embed-facebook-placeholder.png" alt="Embed placeholder Facebook" width="500" />
</p>

<details>
<summary>Example</summary>

```javascript
<template>
  <embed-facebook-placeholder :privacyManagerId="privacyManagerId"></embed-facebook-placeholder>
</template>

<script>
import { EmbedFacebookPlaceholder } from '@spring-media/red-sourcepoint-cmp/dist/esm/vue/components/EmbedFacebookPlaceholder';

export default {
  components: { EmbedFacebookPlaceholder },
  data: () => ({
    privacyManagerId: 12345,
  }),
};
</script>

<style lang="scss">
@import '~@spring-media/red-sourcepoint-cmp/dist/esm/vue/components/EmbedFacebookPlaceholder.css';
</style>
```
</details>

## Props

| Name             | Type   | Required | Description |
| ---------------- | ------ | -------- | ----------- |
| privacyManagerId | number | true     | Id of a privacy manager to open when clicking on certain areas within the placeholder |
