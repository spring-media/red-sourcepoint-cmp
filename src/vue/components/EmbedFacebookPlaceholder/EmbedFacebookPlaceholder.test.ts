import { mount } from '@vue/test-utils';
import EmbedFacebookPlaceholder from './EmbedFacebookPlaceholder.vue';

describe('EmbedFacebookPlaceholder', () => {
  it('should render without any errors', () => {
    expect(mount(EmbedFacebookPlaceholder).element).toMatchSnapshot();
  });
});
