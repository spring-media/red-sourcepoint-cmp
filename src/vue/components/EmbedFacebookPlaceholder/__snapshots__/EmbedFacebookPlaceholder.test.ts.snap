// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmbedFacebookPlaceholder should render without any errors 1`] = `
<div
  class="embed-placeholder__container"
>
  <div
    class="embed-placeholder__header"
  >
    <svg
      fill="none"
      height="84"
      width="100%"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="16"
        cy="16"
        fill="#E9ECEF"
        r="16"
      />
       
      <rect
        fill="#E9ECEF"
        height="10"
        width="60"
        x="40"
        y="7"
      />
       
      <rect
        fill="#E9ECEF"
        height="6"
        width="40"
        x="40"
        y="19"
      />
       
      <rect
        fill="#E9ECEF"
        height="8"
        width="100%"
        y="44"
      />
       
      <rect
        fill="#E9ECEF"
        height="8"
        width="100%"
        y="60"
      />
       
      <rect
        fill="#E9ECEF"
        height="8"
        width="75%"
        y="76"
      />
       
      <path
        d="M20 10.0611C20 4.50451 15.5229 0 10 0C4.47715 0 0 4.50451 0 10.0611C0 15.0828 3.65684 19.2452 8.4375 20V12.9694H5.89844V10.0611H8.4375V7.84452C8.4375 5.32296 9.93047 3.93012 12.2146 3.93012C13.3088 3.93012 14.4531 4.12663 14.4531 4.12663V6.60261H13.1922C11.95 6.60261 11.5625 7.37822 11.5625 8.17387V10.0611H14.3359L13.8926 12.9694H11.5625V20C16.3432 19.2452 20 15.0828 20 10.0611Z"
        fill="#1877F2"
        style="transform: translateX(calc(100% - 20px));"
      />
    </svg>
  </div>
   
  <div
    class="embed-placeholder__headline"
  >
    
    An dieser Stelle findest du Inhalte aus Facebook
  
  </div>
   
  <div
    class="embed-placeholder__description"
  >
    
    Um mit Inhalten aus Facebook und anderen sozialen Netzwerken zu interagieren oder diese darzustellen, brauchen wir
    deine Zustimmung.
  
  </div>
   
  <button
    class="embed-placeholder__button"
  >
    
          soziale Netzwerke aktivieren
        
  </button>
   
  <div
    class="embed-placeholder__footer-text"
  >
    
        Ich bin damit einverstanden, dass mir externe Inhalte aus sozialen Netzwerken angezeigt werden. Damit können
        personenbezogene Daten an Drittanbieter übermittelt werden. Mehr dazu findest du in der
        
    <a
      class="embed-placeholder__text-link embed-placeholder__link-description"
      href="#"
      rel="noopener"
      target="_blank"
    >
      Beschreibung dieses Datenverarbeitungszweck
    </a>
    
        und in den Datenschutzinformationen dieser
        
    <a
      class="embed-placeholder__text-link embed-placeholder__link-vendor-list"
      href="#"
      rel="noopener"
      target="_blank"
    >
      Drittanbieter
    </a>
    .
      
  </div>
</div>
`;
